src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -2491,7 +2491,7 @@ public PSObject EndCreateObject()
     /// Unsafe version of PSObjectBuilder that is not guaranteed to be on the same thread.
     /// Usage must be reviewed carefully.
     /// </summary>
-    internal class PSObjectBuilderUnsafe
+    internal class PSObjectBuilderLessOptimized
```

Comment:
It is not clear - less optimized then what? Should we use full name `PSObjectBuilderLessOptimizedThenVakuePSObjectBuilder` ?
I think we can move `LessOptimized` in comments and use simple name `PSObjectBuilder`.